bigint/src/biguint.rs

```diff
@@ -976,6 +977,17 @@ impl Integer for BigUint {
     }
 }
 
+pub trait ModPow {
+    fn modpow(&self, exponent: &Self, modulus: &Self) -> Self;
```

Comment:
Why is this a trait instead of just adding a method to `BigUint`?

If we really want a trait, it would belong in `num-traits`, hopefully implemented for primitive integers too.  But I think that's unnecessary for the moment.

Reply:
I made a trait because that was how every other mathematical operator was implemented on `BigUint`, so I was trying to follow precedent. I'll move it into `impl BigUint`.